libgnucash/core-utils/gnc-unicode.h

```diff
@@ -0,0 +1,77 @@
+/********************************************************************
+ * gnc-icu-locale.h -- Localization with ICU.                        *
+ *                                                                  *
+ * Copyright (C) 2025 John Ralls <jralls@ceridwen.us                *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ ********************************************************************/
+
+#pragma once
+#include <stdbool.h>
+
+/** @addtogroup Localization These functions perform string comparison
+    and collation according to the Unicode Common Locale Data
+    Repository rules. The CLDR specifies five levels of
+    comparison. The primary or base level considers all variant
+    codepoints representing a character to be equivalent regardless of
```

Comment:
More minor bike shedding: this description calls for a (un)numbered list for readability.